tests/phpunit/Mutator/OperationCategoryTest.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutator;
+
+use function array_keys;
+use function array_unique;
+use function array_values;
+use function in_array;
+use Infection\Mutator\OperationCategory;
+use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+use function Safe\sprintf;
+
+class OperationCategoryTest extends TestCase
+{
+    private const ALL_CONSTANT_KEY = 'ALL';
+
+    public function test_it_cannot_be_instantiated(): void
+    {
+        $classReflection = new ReflectionClass(OperationCategory::class);
+
+        $this->assertFalse(
+            $classReflection->isInstantiable(),
+            sprintf('Did not expect "%s" to be instantiable', OperationCategory::class)
+        );
+    }
+
+    public function test_it_lists_all_its_exposed_constants(): void
```

Comment:
this test might appear as a bit overkill here regarding the assertions. I was a bit lazy and took some elements of a private codebase where we have a simple convention for "enums". Maybe this can be drastically simplified (although it is a bit overkill, it does have the merit of being more readable IMO) and could be re-used for other pseudo-enums we have in the codebase